### PR TITLE
Check condition before dispose "ObjectStreamData"

### PR DESCRIPTION
**Pull request by haiduong87**
We should not dispose external resource.
In this case, we only dispose "ObjectStreamData" when we create it.

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Please fix formatting in the code

```
!!!! Process Aborted !!!!
The following files do not match .editorconfig:
 * Minio/DataModel/ObjectOperationsArgs.cs
Error: Process completed with exit code 1.
```

**Review by ebozduman — APPROVED** [review]
LGTM

File: Minio/DataModel/ObjectOperationsArgs.cs [hunk 1]
```diff
@@ -1891,6 +1891,8 @@ public PutObjectArgs WithStreamData(Stream data)
 
     ~PutObjectArgs()
     {
-        if (ObjectStreamData != null) ObjectStreamData.Close();
+        if (!string.IsNullOrWhiteSpace(FileName))
+            if (ObjectStreamData != null)
```
Comment (ebozduman) [inline Minio/DataModel/ObjectOperationsArgs.cs:1895]:
```suggestion
        if (!string.IsNullOrWhiteSpace(FileName) && ObjectStreamData != null)
```

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: Minio/DataModel/ObjectOperationsArgs.cs (hunk 1 above)
Comment (ebozduman) [reply Minio/DataModel/ObjectOperationsArgs.cs:1895]: @haiduong87 

You've marked this comment as resolved but you did not say why you chose to not include the change here.
Please either explain the reason of your rejection or if you agreed but forgot to make the change, then modify the code.

Comment (haiduong87) [reply Minio/DataModel/ObjectOperationsArgs.cs]: @ebozduman 
I missed this change request.
I just did the "regitlint".

I make new commit to solve your request.

**Review by ebozduman — CHANGES_REQUESTED** [review]
After digging out about the `-1` unknown size paradigm, it looks like that is our approach also in all other SDKs. So, going back to the starting point and @harshavardhana's comment, you need to put the check back again as there is no change is needed.

In other words, this is our design and there is nothing wrong here.

If you would like to upload zero/0 byte files, then you need to set the size to `-1`.

You can close this PR.

**Comment by ebozduman** [comment]
@haidung87

Please look into the build failure

**Comment by haiduong87** [comment]
@ebozduman 
Can you help me here, I run the test locally and it's fine.

**Comment by ebozduman** [comment]
@haiduong87 

Did you run your test in Windows environment?

I also wannted to repro the issue with the failing test, `PutObject_Test8`, but my Windows VM in our data center is down for some reason. I asked someone to check it out.

This change is about garbage disposal, and the failure is about the line you touched in PR #655. I'll look into it and get back to you.

**Comment by haiduong87** [comment]
> > @haiduong87
> > Did you run your test in Windows environment?   <<<===
> > I also wannted to repro the issue with the failing test, `PutObject_Test8`, but my Windows VM in our data center is down for some reason. I asked someone to check it out.
> > This change is about garbage disposal, and the failure is about the line you touched in PR #655. I'll look into it and get back to you.
> 
> @haiduong87
> 
> Could you reply my question above (<<<===)? I am on hold waiting for my Windows VM setup in our data center to be looked into for me to try reproducing the build issue.

@ebozduman 
I'm a Windows user. I did run the test on Windows environment

**Comment by haiduong87** [comment]
@ebozduman 

I think this error is not related to https://github.com/minio/minio-dotnet/pull/655

And this PR was successfully tested in widows environment (in previous flows)

**Comment by ebozduman** [comment]
@haiduong87 

Could you rebase your branch ?

**Comment by haiduong87** [comment]
@ebozduman 
I merge 'master' into patch-1

**Review by harshavardhana — APPROVED** [review]

**Comment by ebozduman** [comment]
I've tried to reproduce the issue on a VM (Windows server 2019) in our data center, but could not reproduce it. 

I'll be still looking into the issue.

**Review by ebozduman — APPROVED** [review]

**Pull request merged**